com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/query/providers/SqlFileQueryProvider.java

```diff
@@ -63,7 +63,11 @@ public SqlFileQueryProvider(String tableName, DataSourceDialect dataSourceDialec
                 statement.setString(7, fileEntry.getDigestAlgorithm());
                 statement.setTimestamp(8, new Timestamp(fileEntry.getModified()
                     .getTime()));
-                getDataSourceDialect().setBlobAsBinaryStream(statement, 9, content);
+                if (isContentStoredAsBlob) {
+                    getDataSourceDialect().setBlobAsBinaryStream(statement, 9, content);
```

Comment:
Make the class abstract with the two abstract methods for getting and storing content. Use one of them in place of this if statement.